frontend_lib/src/component/EmojiReactions/EmojiReactionButtons.jsx

```diff
@@ -0,0 +1,252 @@
+              placement='left'
+              isOpen
+              target={() => this.buttonRef.current}
+            >
+              <PopoverBody>
+                <Picker
+                  native
+                  emojiTooltip={true}
+                  title=''
+                  i18n={emojiMartI18n}
+                  onSelect={this.handleSelect}
+                />
+              </PopoverBody>
+            </UncontrolledPopover>
+          )
+          : undefined
+        )}
+      </>
+    )
+  }
+}
+
+const TranslatedEmojiPickerButton = translate()(EmojiPickerButton)
+
+function getReactionButtonTitle (props) {
+  const reactionList = props.reactionList
+
+  const you = (
+    reactionList[0].reaction_id === props.userReactionId
+      ? props.t('You')
+      : props.t('you')
+  )
+
+  const value = props.reactionList[0].value
+  const emoji = getEmojiDataFromNative(value, 'apple', emojiData)
+
+  const users = andList(
+    props.reactionList.map(
+      ({ author, reaction_id: reactionId }) => (
+        reactionId === props.userReactionId
+          ? you
+          : (author.public_name || author.username)
+      )
+    )
+  )
+
+  const l10nOpts = {
+    reaction: emoji ? (emoji.id || emoji.name) : value,
+    users: users,
+    user: users
+  }
+
+  return (
+    props.userReactionId === -1
+      ? (
+        props.reactionList.length === 1
+          ? props.t('{{user}} reacted with {{reaction}}', l10nOpts)
+          : props.t('{{users}} reacted with {{reaction}}_them', l10nOpts)
+      )
+      : (
+        props.reactionList.length === 1
+          ? props.t('You reacted with {{reaction}}', l10nOpts)
+          : props.t('{{users}} reacted with {{reaction}}_you', l10nOpts)
+      )
+  )
+}
+
+const ReactionButton = translate()(function ReactionButton (props) {
+  const highlighted = props.userReactionId !== -1
+  const title = getReactionButtonTitle(props)
+
+  return (
+    <button
+      onClick={highlighted ? props.onRemoveReaction : props.onAddReaction}
+      title={title}
+      className={classnames('ReactionButton__button', { highlighted })}
+    >
+      <span className='ReactionButton__button__value'>{props.reactionList[0].value}</span>
+      <span className='ReactionButton__button__count'>{props.reactionList.length}</span>
+    </button>
+  )
+})
+
+ReactionButton.propTypes = {
+  onRemoveReaction: PropTypes.func.isRequired,
+  onAddReaction: PropTypes.func.isRequired
+}
+
+const EmojiReactionButtons = (props) => {
+  const reactionListByValue = {}
+
+  for (const reaction of props.reactionList) {
+    const bucket = (
+      reactionListByValue[reaction.value] || (
+        reactionListByValue[reaction.value] = {
+          firstCreation: reaction.created,
+          reactionList: []
+        }
+      )
+    )
+
+    bucket.reactionList.push(reaction)
+
+    if (!bucket.firstCreation || bucket.firstCreation > reaction.created) {
+      bucket.firstCreation = reaction.created
+    }
+  }
```

Comment:
I think you should add a comment to explain what this does: grouping reactions per value.

Reply:
Ok, I put that into a function that has the good taste of having a name